charts/karpenter/README.md

```diff
@@ -45,6 +45,7 @@ helm upgrade --install --namespace karpenter --create-namespace \
 | controller.logLevel | string | `""` | Controller log level, defaults to the global log level |
 | controller.resources | object | `{"limits":{"cpu":1,"memory":"1Gi"},"requests":{"cpu":1,"memory":"1Gi"}}` | Resources for the controller pod. |
 | controller.securityContext | object | `{}` | SecurityContext for the controller container. |
+| deploymentAdditionalAnnotations | object | `{"kubectl.kubernetes.io/default-container":"controller","kubectl.kubernetes.io/default-logs-container":"controller"}` | Additional annotations to add into deployment's metadata. |
```

Comment:
We should just bake this directly into to deployment.yaml instead of passing it through.

Reply:
Yeah I thought about this but I remember that, as a Helm user, there is nothing that I hate more than when there is hardcoded properties in the templates that I can't modify without forking the chart.

Reply:
I think this annotation is a little different since you shouldn't ever modify it. I would also advocate for this to be hardcoded in the `deployment.yaml`.

For other annotations you'd like to add, you still have the option to add to the `additionalAnnotations` field, even if it adds the annotations to more than just the deployment resource :)

Reply:
I agree, there are the `additionalAnnotations` or `podAnnotations` value to pass any custom annotations into the chart.